### ISSUE TITLE
Add a --not-a-security-boundary option

### ISSUE THREAD

**Issue by smcv** [containers/bubblewrap#653]
Depending on the options that it's given, the new namespaces created by bubblewrap might be:

* a sandboxed environment, with a security boundary between the real system and the sandbox (for example Flatpak uses it this way when running the actual app)
* a non-sandboxed environment, with differences between the real system and the sandbox, but no meaningful security boundary (for example Flatpak uses it this way when running `xdg-dbus-proxy`, and the Steam Runtime uses it this way when running Steam games)

If there is meant to be a security boundary, lots of features of bubblewrap need to "fail closed". For example, if we are told to provide a filesystem, and we fail to remount it `ro` or `nosuid`, then our only choice is to refuse to run the wrapped program (fail closed), because if we carried on anyway (fail open) then we would not be providing the intended security properties.

However, if there is not meant to be a security boundary, for a lot of failures we can consider ignoring the failure (possibly with a warning) and carrying on anyway.

At the moment, bubblewrap has no way to tell which of these security models the user of bubblewrap is looking for, so we have to make the conservative/pessimistic assumption that they *do* want a security boundary, and fail closed.

However, if we added a `--not-a-security-boundary` option, we could use it to make bubblewrap more robust in the cases where something unexpected goes wrong, such as #650. Pseudocode:

```
if (a thing fails)
  {
    if (not_a_security_boundary)
      warn ("the thing failed (%s), continuing without doing the thing", strerror (errno));
    else
      die_with_error ("the thing failed");
  }
```

**Comment by smcv** [comment]
cc @refi64

**Comment by xplshn** [comment]
This would be a desired feature for a large part of the AppImages community, AppBundle community, NixAppImage project, and others. It'd be great to have this :)

**Comment by xplshn** [comment]
@smvc meanwhile, could a list of bwrap flags necessary to make it work like a simple `chroot` that doesn't sandbox anything be provided? I've been battling with `bwrap` for various hours now, I can't figure out how to make it disable all sandboxing capabilities, including but not limited to seccomp, etc.

![Image](https://github.com/user-attachments/assets/9f608d81-7d77-4a9d-b464-f68f331643a9)
![Image](https://github.com/user-attachments/assets/6ca9b156-d8cb-4912-8a96-1f8035bffe9d)

It'd be great if at least a list of these were provided

**Comment by xplshn** [comment]
> bwrap --dev-bind /chroot / -- COMMAN

Thanks! I'll be using that

**Comment by smcv** [comment]
As well as mitigating https://github.com/containers/bubblewrap/issues/650, this would likely be useful for coping gracefully with automounts, for example:

* https://github.com/flatpak/flatpak/issues/5112
* https://github.com/ValveSoftware/steam-runtime/issues/766
* https://github.com/ValveSoftware/steam-for-linux/issues/10571

**Comment by ao2** [comment]
A draft implementation of `--not-a-security-boundary` can be found in https://github.com/containers/bubblewrap/pull/751

**Comment by ao2** [comment]
This change should also address https://github.com/containers/bubblewrap/issues/541

**Comment by twhitehead** [comment]
@smcv would it make sense for this option to also not set `PR_SET_NO_NEW_PRIVS`? In particular I am thinking of executing programs that have `cap_sys_nice` set on them (in order for them to be able to set real time priority for some of their threads) not working in bwrap.

For example, adding `cap_sys_nice` to `chrt` to demonstrate this
```
cp $(which chrt) .
sudo setcap cap_sys_nice=eip chrt
```
It runs okay outside of `bwrap`
```
./chrt 99 echo hi
```
```
hi
```
but fails inside of `bwrap`
```
bwrap --ro-bind / / --dev /dev --proc /proc --tmpfs /tmp ./chrt 99 echo hi
```
```
chrt: failed to set pid 0's policy: Operation not permitted
```

**Comment by smcv** [comment]
> would it make sense for this option to also not set `PR_SET_NO_NEW_PRIVS`?

No. In the presence of user namespaces, setcap can't work the way you expect it to anyway.

> In particular I am thinking of executing programs that have `cap_sys_nice` set on them (in order for them to be able to set real time priority for some of their threads) not working in bwrap

This can't work the way you hope it does. The kernel's data model is that each process can only have capabilities in the "innermost" user namespace; and to be able to do its job, bubblewrap needs to create a new user namespace.

This means that the most-capable your process inside the container can possibly be is to have `CAP_SYS_NICE` in the user namespace that was created by bubblewrap. It cannot have `CAP_SYS_NICE` in the "init namespace" (the namespace where the kernel started init), which is the requirement for being allowed to set real-time priority on yourself.

Instead, you'll need to send D-Bus messages to [RealtimeKit](https://gitlab.freedesktop.org/pipewire/rtkit/), the [Realtime portal](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.Realtime.html) or a similar service outside the container, and ask it to give your threads real-time priority "from outside".
